src/components/JoinPage/EmailConfirm/index.tsx

```diff
@@ -0,0 +1,39 @@
+import { useLocation, useNavigate } from "react-router-dom";
+
+import { prevPages } from "../../../core/join/prevPages";
+import { progressRate } from "../../../core/join/progressRate";
+import { routePaths } from "../../../core/routes/path";
+import Footer from "../../common/Footer";
+import Header from "../common/Header";
+import PageProgressBar from "../common/PageProgressBar";
+import { St } from "./style";
+
+export default function EmailConfirm() {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const state = location.state as { userEmail: string };
+  const userEmail = state.userEmail;
```

Comment:
```suggestion
  const { state: { useEmail } } = useLocation();
```

이런 식으로 줄일 수 있지 않을까 합니다! (+ 타입단언)

Reply:
까리하다 주함... 감쟈해욤💭